### PR TITLE
Some small code improvements

### PR DESCRIPTION
**Pull request by MichelBoucey**
Hi,

[For a better Haskell style](https://github.com/tibbe/haskell-style-guide/blob/master/haskell-style.md).

### CODE REVIEW

**Comment by MichelBoucey** [comment]
This travis build target, "BUILD=cabal GHCVER=7.8.4" in .travis.yml, has failed because of the so-called AMP transition, since GHC 7.10.

**Comment by richardTowers** [comment]
I guess we can just drop support for GHC 7.8.4 until someone says they need it.

I'll remove it from the build matrix and peg the GHC version to >=7.10.3.

**Comment by MichelBoucey** [comment]
You're right.

I guess that >=7.10.1 will be fine.

File: stack.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
-resolver: lts-5.15
+flags: {}
+extra-package-dbs: []
 packages:
 - '.'
 extra-deps:
-- wai-extra-3.0.15.1
-flags: {}
-extra-package-dbs: []
+- OneTuple-0.2.1
```
Comment (richardTowers) [inline stack.yaml:6]: Do we actually need OneTuple? Or is this just an accidental?

Comment (MichelBoucey) [inline stack.yaml:6]: It's a dependency of the tuple package.

**Comment by richardTowers** [comment]
Thanks for this, merged.

**Comment by MichelBoucey** [comment]
Ok.

BTW : always run [hlint](https://hackage.haskell.org/package/hlint) on your Haskell code for hints.

**Comment by richardTowers** [comment]
Thanks, I'll add it to my editor / stick it in `make test`.

**Pull request merged**